tools/codegen/pkg/interfacegen/generator_test.go

```diff
@@ -15,159 +15,89 @@
+func TestGenerate_Errors(t *testing.T) {
+	g := Generator{OutFilePath: "."}
+	err := g.Generate("testdata/error_template.descriptor_set")
+	if err == nil {
+		t.Fatalf("Generate(%s) should have produced an error", "testdata/error_template.descriptor_set")
 	}
-	cmd := exec.Command("protoc", protocCmd...)
-	dir := path.Join(os.Getenv("GOPATH"), "src/istio.io")
-	cmd.Dir = dir
-	cmd.Stderr = os.Stderr // For debugging
-	err := cmd.Run()
-	return err
-}
-
-// TODO: This is blocking the test to be enabled from Bazel.
-func generteGoPbFileForTmpl(protoFile string, outDir string) error {
-
-	// HACK HACK. Depending on dir structure is super fragile.
-	// Explore how to generate File Descriptor set in a better way.
-	protocCmd := []string{
-		path.Join("mixer/tools/codegen/pkg/interfacegen", protoFile),
-		"--go_out=Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,Mmixer/v1/" +
-			"config/descriptor/value_type.proto=istio.io/api/mixer/v1/config/descriptor," +
-			"Mgoogle/protobuf/descriptor.proto=github.com/golang/protobuf/protoc-gen-go/descriptor," +
-			"Mgoogle/protobuf/struct.proto=github.com/golang/protobuf/ptypes/struct," +
-			"Mmixer/tools/codegen/pkg/template_extension/TemplateExtensions.proto=istio.io/mixer/tools/codegen/pkg/template_extension:.",
-		"-I=.",
-		"-I=api",
+	b, fileErr := ioutil.ReadFile("testdata/ErrorTemplate.baseline")
+	if fileErr != nil {
+		t.Fatalf("Could not read baseline file: %v", err)
 	}
+	want := fmt.Sprintf("%s", b)
+	got := err.Error()
+	if got != want {
+		t.Fatalf("Generate(%s) => '%s'\nwanted: '%s'", "testdata/error_template.descriptor_set", got, want)
+	}
+}
 
-	cmd := exec.Command("protoc", protocCmd...)
-	dir := path.Join(os.Getenv("GOPATH"), "src/istio.io")
-	cmd.Dir = dir
-	cmd.Stderr = os.Stderr // For debugging
-	err := cmd.Run()
+const chunkSize = 64000
 
+func deepCompare(file1, file2 string, logf logFn) bool {
+	f1, err := os.Open(file1)
 	if err != nil {
-		return err
+		logf("could not open file: %v", err)
+		return false
 	}
 
-	// Copy to the outDir
-	genGoFilePath := path.Join(path.Dir(protoFile), getBaseFileNameWithoutExt(protoFile)+".pb.go")
-	// first do the magic replacement
-	sedCmd := exec.Command("sed", "-i", "-e", "s/ValueType_VALUE_TYPE_UNSPECIFIED/VALUE_TYPE_UNSPECIFIED/g", genGoFilePath)
-	sedCmd.Stdout = os.Stdout
-	sedCmd.Stderr = os.Stderr
-	err = sedCmd.Run()
+	f2, err := os.Open(file2)
 	if err != nil {
-		return err
+		logf("could not open file: %v", err)
+		return false
 	}
 
-	mvCmd := exec.Command("mv", genGoFilePath, outDir)
-	mvCmd.Stdout = os.Stdout
-	mvCmd.Stderr = os.Stderr
+	for {
+		b1 := make([]byte, chunkSize)
+		s1, err1 := f1.Read(b1)
 
-	return mvCmd.Run()
-}
+		b2 := make([]byte, chunkSize)
+		s2, err2 := f2.Read(b2)
 
-func getBaseFileNameWithoutExt(filePath string) string {
-	tmp := filepath.Base(filePath)
-	return tmp[0 : len(tmp)-len(filepath.Ext(tmp))]
+		if err1 != nil || err2 != nil {
+			if err1 == io.EOF && err2 == io.EOF {
```

Comment:
can this be just 
s2, err2 := f2.Read(b2)
if err1 == io.EOF && err2 == io.EOF {
  return true
}
if !bytes.Equal(b1, b2) {
	logf("bytes don't match (sizes: %d, %d):\n%s\n%s", s1, s2, string(b1), string(b2))
	return false
}